### PR TITLE
Expand dependency version support in Icon Transformer

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

Reported in Slack

For the Icon builder
Run verdaccio to publish, then use npx to run the cli in some other context that isn't our own repo

For the Transformer
Run in a parcel project that uses the parcel 2.12 engine, or anything other than the dev version we used to be listed at here

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

**Review by akyllus — APPROVED** [review]

**Review by akyllus — APPROVED** [review]

File: packages/dev/parcel-transformer-s2-icon/package.json [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
   "main": "IconTransformer.js",
   "engines": {
-    "parcel": "2.0.0-dev.1599"
+    "parcel": "^2.0.0"
   },
   "dependencies": {
     "@adobe/spectrum-tokens": "^13.0.0-beta.46",
-    "@parcel/plugin": "2.0.0-dev.1601",
-    "@svgr/core": "^6.2.0"
+    "@parcel/plugin": "^2.0.0-dev.1601",
```
Comment (devongovett) [inline packages/dev/parcel-transformer-s2-icon/package.json:10]: shouldn't this change as well?

File: scripts/verdaccio.sh [hunk 2]
```diff
@@ -161,6 +161,12 @@ then
   yarn build
   mv dist ../../$verdaccio_path/s2-webpack-5-example
 
+  echo 'test icon builder'
+  cd ../../examples/s2-webpack-5-example
+  mkdir icon-test
```
Comment (devongovett) [inline scripts/verdaccio.sh:166]: Isn't this still technically testing within our repo? i.e. `../../node_modules` will exist (outside examples folder). So even if the dependencies were missing they'd still get picked up here?

Comment (LFDanLu) [reply scripts/verdaccio.sh:166]: Right, this is the reason why running the icon builder inside our repo works (we have svgo as part of a different dependency) but not outside our repo if I understand the original issue correctly

**Review by LFDanLu — COMMENTED** [review]
Verified locally via verdaccio and a random project I had locally, verified that the icon builder successfully ran and created the proper tsx file from the provided svg.

**Review by devongovett — APPROVED** [review]
We should ensure that the verdaccio script is really testing outside the repo in a followup

**Review by reidbarber — APPROVED** [review]

**Pull request merged**